microprofile/server/src/main/java/io/helidon/microprofile/server/ServerImpl.java

```diff
@@ -446,14 +447,13 @@ public Server start() {
         CountDownLatch cdl = new CountDownLatch(1);
         AtomicReference<Throwable> throwRef = new AtomicReference<>();
 
-        long beforeT = System.nanoTime();
         server.start()
                 .whenComplete((webServer, throwable) -> {
                     if (null != throwable) {
                         STARTUP_LOGGER.log(Level.FINEST, "Startup failed", throwable);
                         throwRef.set(throwable);
                     } else {
-                        long t = TimeUnit.MILLISECONDS.convert(System.nanoTime() - beforeT, TimeUnit.NANOSECONDS);
+                        long t = TimeUnit.MILLISECONDS.convert(System.nanoTime() - STARTUP_TIME, TimeUnit.NANOSECONDS);
```

Comment:
Isn't it technically possible for a server to be started, stopped and started again? I think we may need to redefine what we call _startup time_ here. It should really be the elapsed time of calling `server.start()`, trying to calculate anything else in this method would be inaccurate; any such calculation needs to happen elsewhere IMO.

Reply:
I'd originally noticed that, after some code was added to the `start` method, the point where the start time was captured fell _after_ that added work. In the slack discussion  Tomas expressed the opinion that the reported time should include as much work as possible -- he mentioned CDI + the server start time.

Yes, it is conceivable that a server would be started, stopped, and restarted again although that's not the usual use case. Recording the system time at the very beginning of the `start` method results in a reported start-up time of 100 to 300 ms (in informal experiments). That's significantly under the log message timestamp delta, and that kind of mismatch is what prompted this issue originally.

The CDI initialization occurs in `Server.Builder.build()` (unless the caller to `Builder` passes a previously-init'd CDI container which is not our usual case). 

We could separately record the "initialization" time (which would include the CDI initialization) and the server start-up time. During the first start-up the log message could report the overall time and the two components; during a subsequent server restart it could report only the time in `ServerImpl.start` because the other initialization is not repeated in that case.

Reply:
Or, because the server start-up time itself seems to be a relatively small part of the overall initialization time, we could report the overall initialization time only once, at the end of the first invocation of `start`. Rereading what I wrote above it might be overly confusing to users to try to concisely convey the two separate times.